pkg/k8shandler/visualization.go

```diff
@@ -165,6 +167,9 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateKibanaDeployment() (e
 		"kibana",
 		kibanaPodSpec,
 	)
+	kibanaDeployment.ObjectMeta.Annotations = map[string]string{
```

Comment:
this logic isn't enough, we don't check annotation values at all while determining if an update is required (in `isKibanaDifferent`)

Reply:
This is still missing

Reply:
Does this not resolve your ask: https://github.com/openshift/cluster-logging-operator/pull/217/files#diff-5ff1cbe659b99e0e73d8ba484249c27cR579

Reply:
i was thrown off by the function name... can we abstract the logic in `AreSelectorsSame` to something like `areMapsSame` and have a `AreSelectorsSame` and `AreAnnotationsSame` and call `areMapsSame` underneath that?

Reply:
@ewolinetz are you agreeable to skipping the layering and just using `AreMapsSame`?  Is there a reason we cannot use `reflect.DeepEquals` and remove the function all together?

Reply:
reflect is the recommended solution: https://stackoverflow.com/questions/18208394/how-to-test-the-equivalence-of-maps-in-golang

Reply:
I'm using `reflect.DeepEqual` in the rsyslog undefined_field tests: https://github.com/openshift/origin-aggregated-logging/blob/master/rsyslog/undefined_field/undefined_field_test.go#L279

Reply:
yeah, i had tried to use `reflect` before but found it didn't correctly evaluate which is why its a function instead... :man_shrugging: